CMakeLists.txt

```
@@ -41,15 +41,44 @@ FIND_PACKAGE( DD4hep REQUIRED )
 FIND_PACKAGE( ROOT REQUIRED )
 FIND_PACKAGE( MarlinTrk REQUIRED )
 
+OPTION( USE_SDHCALCONTENT "Use SDHCALContent" OFF )
+OPTION( USE_APRILCONTENT "Use APRILContent" OFF )
 
-FOREACH( pkg PandoraSDK Marlin MarlinUtil  LCContent DD4hep ROOT MarlinTrk )
+FOREACH( pkg PandoraSDK Marlin MarlinUtil LCContent DD4hep ROOT MarlinTrk )
     IF( ${pkg}_FOUND )
         INCLUDE_DIRECTORIES( SYSTEM ${${pkg}_INCLUDE_DIRS} )
         LINK_LIBRARIES( ${${pkg}_LIBRARIES} )
         ADD_DEFINITIONS ( ${${pkg}_DEFINITIONS} )
     ENDIF()
 ENDFOREACH()
 
+IF( USE_SDHCALCONTENT )
+    FIND_PACKAGE( SDHCALContent REQUIRED )
+    IF( SDHCALContent_FOUND )
+        INCLUDE_DIRECTORIES( SYSTEM ${SDHCALContent_INCLUDE_DIRS} )
+        LINK_LIBRARIES( ${SDHCALContent_LIBRARIES} )
+        ADD_DEFINITIONS ( ${SDHCALContent_DEFINITIONS} )
+        ADD_DEFINITIONS( "-DSDHCALCONTENT" )
+    ENDIF()
```

```suggestion
    INCLUDE_DIRECTORIES( SYSTEM ${SDHCALContent_INCLUDE_DIRS} )
    LINK_LIBRARIES( ${SDHCALContent_LIBRARIES} )
    ADD_DEFINITIONS ( ${SDHCALContent_DEFINITIONS} )
    ADD_DEFINITIONS( "-DSDHCALCONTENT" )
```
If you have `REQUIRED` in your `find_package` call there is no need to check again if it is actually found, because CMake will actually stop if it is not found.

---

```suggestion
    IF( SDHCALContent_FOUND )
        INCLUDE_DIRECTORIES( SYSTEM ${SDHCALContent_INCLUDE_DIRS} )
        LINK_LIBRARIES( ${SDHCALContent_LIBRARIES} )
        ADD_DEFINITIONS ( ${SDHCALContent_DEFINITIONS} )
        ADD_DEFINITIONS( "SDHCALCONTENT" )
    ENDIF()
```

cmake will add `-D` when needed, and in places where we have to treat definitions programatically it is easier if we don't have to deal with some with -D and some without.

---

Same as above